ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -355,6 +355,38 @@ PrjFS_Result PrjFS_WritePlaceholderFile(
     return PrjFS_Result_EIOError;
 }
 
+PrjFS_Result PrjFS_WriteSymLink(
```

Comment:
I'm wondering if we actually need symlink methods in our API? I don't think it hurts anything, and we can always think about cleaning this up later, but just to make sure I'm understanding it properly - there's no special logic in here related to our placeholder formats or anything, right? The provider could just as easily create a symlink, or delete a placeholder to replace with a symlink, right?

Reply:
>there's no special logic in here related to our placeholder formats or anything, right? The provider could just as easily create a symlink, or delete a placeholder to replace with a symlink, right?

I had just written up a response that you're correct and that I only did this for API consistency but I then realized that we do need some special logic here.

After the symlink is created it should be flagged as being in the root.  I will make that change (and it's one more reason for keeping these symlink APIs in our library).

Reply:
Ah yea that's a good point. In fact, the provider MUST NOT create any sort of file on its own, if it expects to ever get back notifications on those files.

I can see two ways of dealing with this, though all of this is out of scope for this PR:
1) panic if the provider creates a new file on its own
2) rather than early-existing our kext on all IO from the provider, handle some of the events that are safe. E.g. new file events need to get sent up to the lib so the files can get stamped with the right flags, but the lib could be smart about not forwarding that on to the provider if the provider itself was the cause of the event. And now that I've typed that out, it feels like a can of worms and option 1 may be the safer one :-)